File: actblelibrary/src/main/java/com/yeolabgt/mahmoodms/actblelibrary/ActBleCharacteristic.java
package com.yeolabgt.mahmoodms.actblelibrary;

import android.bluetooth.BluetoothGatt;
import android.bluetooth.BluetoothGattCharacteristic;
import android.bluetooth.BluetoothGattDescriptor;

/**
 * Created by mmahmood31 on 11/8/2017.
 * Object for processing requests upon (implicitly) readable or writable characteristics
 */

public class ActBleCharacteristic {
    private BluetoothGatt bluetoothGatt;
    private int requestCode;
    private BluetoothGattCharacteristic bluetoothGattCharacteristic = null;
    private BluetoothGattDescriptor bluetoothGattDescriptor = null;
    private boolean operationSuccess = false;

    public ActBleCharacteristic(int requestCode, BluetoothGatt bluetoothGatt,
                                BluetoothGattCharacteristic bluetoothGattCharacteristic) {
        this.bluetoothGatt = bluetoothGatt;
        this.requestCode = requestCode;
        this.bluetoothGattCharacteristic = bluetoothGattCharacteristic;
    }

    public ActBleCharacteristic(int requestCode, BluetoothGatt bluetoothGatt,
                                BluetoothGattDescriptor bluetoothGattDescriptor, BluetoothGattCharacteristic bluetoothGattCharacteristic) {
        this.bluetoothGatt = bluetoothGatt;
        this.requestCode = requestCode;
        this.bluetoothGattDescriptor = bluetoothGattDescriptor;
        this.bluetoothGattCharacteristic = bluetoothGattCharacteristic;
    }

    public int getRequestCode() {
        return requestCode;
    }

    public BluetoothGatt getBluetoothGatt() {
        return bluetoothGatt;
    }

    public BluetoothGattCharacteristic getBluetoothGattCharacteristic() {
        return bluetoothGattCharacteristic;
    }

    public void setBluetoothGattCharacteristic(BluetoothGattCharacteristic
                                                       bluetoothGattCharacteristic) {
        this.bluetoothGattCharacteristic = bluetoothGattCharacteristic;
    }

    public BluetoothGattDescriptor getBluetoothGattDescriptor() {
        return bluetoothGattDescriptor;
    }

    public void setBluetoothGattDescriptor(BluetoothGattDescriptor
                                                   bluetoothGattDescriptor) {
        this.bluetoothGattDescriptor = bluetoothGattDescriptor;
    }

    public boolean isOperationSuccess() {
        return operationSuccess;
    }

    public void setOperationSuccess(boolean operationSuccess) {
        this.operationSuccess = operationSuccess;
    }
}

File: actblelibrary/src/main/java/com/yeolabgt/mahmoodms/actblelibrary/ActBleCharacteristic.kt
package com.yeolabgt.mahmoodms.actblelibrary

import android.bluetooth.BluetoothGatt
import android.bluetooth.BluetoothGattCharacteristic
import android.bluetooth.BluetoothGattDescriptor

/**
 * Created by mmahmood31 on 11/8/2017.
 * Object for processing requests upon (implicitly) readable or writable characteristics
 */

class ActBleCharacteristic {
    var bluetoothGatt: BluetoothGatt? = null
        private set
    var requestCode: Int = 0
        private set
    var bluetoothGattCharacteristic: BluetoothGattCharacteristic? = null
        private set
    var bluetoothGattDescriptor: BluetoothGattDescriptor? = null

    constructor(requestCode: Int, bluetoothGatt: BluetoothGatt,
                bluetoothGattCharacteristic: BluetoothGattCharacteristic) {
        this.bluetoothGatt = bluetoothGatt
        this.requestCode = requestCode
        this.bluetoothGattCharacteristic = bluetoothGattCharacteristic
    }

    constructor(requestCode: Int, bluetoothGatt: BluetoothGatt,
                bluetoothGattDescriptor: BluetoothGattDescriptor, bluetoothGattCharacteristic: BluetoothGattCharacteristic) {
        this.bluetoothGatt = bluetoothGatt
        this.requestCode = requestCode
        this.bluetoothGattDescriptor = bluetoothGattDescriptor
        this.bluetoothGattCharacteristic = bluetoothGattCharacteristic
    }

}
